rebuild/src/public_key.h

```diff
@@ -0,0 +1,96 @@
+#ifndef BLST_TS_PUBLIC_KEY_H__
+#define BLST_TS_PUBLIC_KEY_H__
+
+#include <memory>
+
+#include "addon.h"
+#include "blst.hpp"
+#include "napi.h"
+
+#define BLST_TS_PUBLIC_KEY_LENGTH_COMPRESSED 48U
+#define BLST_TS_PUBLIC_KEY_LENGTH_UNCOMPRESSED 96U
+
+typedef struct {
+    std::unique_ptr<blst::P1_Affine> smart_pointer;
+    blst::P1_Affine *raw_point;
+} P1AffineGroup;
+
+class P1Wrapper {
+   public:
+    virtual ~P1Wrapper() = default;
+    virtual bool IsInfinite() = 0;
+    virtual bool InGroup() = 0;
+    virtual void Serialize(bool compress, blst::byte *out) = 0;
+    virtual void AddTo(blst::P1 &point) = 0;
+    virtual blst::P1 MultiplyBy(
+        blst::byte *rand_bytes, size_t rand_bytes_length) = 0;
+    virtual P1AffineGroup AsAffine() = 0;
+};
+
+class P1 : public P1Wrapper {
+    blst::P1 _point;
+
+   public:
+    P1(blst::P1 point) : _point(std::move(point)) {}
+    bool IsInfinite() override { return _point.is_inf(); }
+    bool InGroup() override { return _point.in_group(); }
+    void Serialize(bool compress, blst::byte *out) override {
+        compress ? _point.compress(out) : _point.serialize(out);
+    }
+    void AddTo(blst::P1 &point) override { point.add(_point); };
+    blst::P1 MultiplyBy(
+        blst::byte *rand_bytes, size_t rand_bytes_length) override {
+        blst::byte out[96];
```

Comment:
I guess that's `BLST_TS_PUBLIC_KEY_LENGTH_UNCOMPRESSED`?

Reply:
oops.  missed those.  updated in f8c1239806775ab9e6e635cf11a2cd7fb580ef21